app/src/main/kotlin/com/simplemobiletools/clock/services/timerService.kt

```diff
@@ -0,0 +1,158 @@
+import org.greenrobot.eventbus.ThreadMode
+
+fun Context.startTimerService() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        startForegroundService(Intent(this, TimerService::class.java))
+    } else {
+        startService(Intent(this, TimerService::class.java))
+    }
+}
+
+class TimerService : Service() {
+
+    private var timer: CountDownTimer? = null
+    private var lastTick = 0L
+    private val bus = EventBus.getDefault()
+
+    override fun onCreate() {
+        super.onCreate()
+        bus.register(this)
+    }
+
+    override fun onBind(intent: Intent?): IBinder? = null
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        super.onStartCommand(intent, flags, startId)
+
+        val formattedDuration = config.timerSeconds.getFormattedDuration()
+        startForeground(TIMER_RUNNING_NOTIF_ID, notification(formattedDuration))
+
+        when (val state = config.timerState) {
+            is TimerState.Idle -> bus.post(TimerState.Start(config.timerSeconds.secondsToMillis))
+            is TimerState.Paused -> bus.post(TimerState.Start(state.tick))
+            is TimerState.Running -> bus.post(TimerState.Pause(state.tick))
+            else -> {}
+        }
+
+        return START_NOT_STICKY
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onMessageEvent(state: TimerState.Idle) {
+        config.timerState = state
+        timer?.cancel()
+        stopService()
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onMessageEvent(state: TimerState.Start) {
+        timer = object : CountDownTimer(state.duration, 1000) {
+            override fun onTick(tick: Long) {
+                lastTick = tick
+
+                val newState = TimerState.Running(state.duration, tick)
+                bus.post(newState)
+                config.timerState = newState
+            }
+
+            override fun onFinish() {
+                bus.post(TimerState.Finish(state.duration))
+            }
+        }.start()
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onMessageEvent(event: TimerState.Finish) {
+        val pendingIntent = getOpenTimerTabIntent()
+        val notification = getTimerNotification(pendingIntent, false) //MAYBE IN FUTURE ADD TIME TO NOTIFICATION
+        val notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        notificationManager.notify(TIMER_NOTIF_ID, notification)
+
+        bus.post(TimerState.Idle)
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onMessageEvent(event: TimerState.Pause) {
+        bus.post(TimerState.Paused(event.duration, lastTick))
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onMessageEvent(state: TimerState.Paused) {
+        config.timerState = state
+        timer?.cancel()
+        stopService()
+    }
+
+    private fun stopService() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) stopForeground(true)
```

Comment:
missing brackets, split it into multiple lines. There is also an isOreoPlus helper function, use that